d2-schemas/src/main/pegasus/com/linkedin/d2/D2RelativeStrategyProperties.pdl

```diff
@@ -104,14 +73,23 @@ record D2RelativeStrategyProperties {
   slowStartThreshold: optional double
 
   /**
-   * Regular expression used to match a request status code and indicate a server-side error.
+   * A request status code that falls within any of the provided ranges will be considered as a server-side error.
    */
-  errorStatusRegex: optional string
+  errorStatusFilter: optional array[HttpStatusCodeRange]
 
   /**
    * Emitting interval (in milliseconds) for D2Events.
    */
-  emittingIntervalMs: optional int
+  emittingIntervalMs: optional long
+
+  /**
+   * When enabled, fully degraded hosts will "recover" health despite not taking any traffic.
+   * The health recovery will model that of slow start. Health score will be doubled as long
+   * as the number of requests toward the entire cluster exceed the minimum call count (minCallCount).
+   * The idea is to prevent unhealthy hosts from never receiving traffic and being able to
+   * naturally recover due to their health score and overall QPS to the cluster being too low.
+   */
+  enableFastRecovery: optional boolean
```

Comment:
Any reason we want to expose this config to user (through PDL)? This should be an internal knob that we only set by program. Exposing it to users can cause unnecessary confusion.

Reply:
Originally we have a field that is more like a switch for fast recovery feature: ringRampFactor, the default is 1.0.
Now we removed the ringRampFactor, and replaced with this boolean, and always use 2.0 as the recovery rate if user wants to enable the fast recovery.

Reply:
As Ruxin said, this feature was already opt-in by configuring ringRampFactor (although the default is opt-in). Now we've made it more clear with this config. If you'd like we can remove it. I'm okay with either way.

Reply:
ah I see. We can keep it in this case.